### DISCUSSION TITLE
Request: add support for Cerebras GPT models just released

### DISCUSSION THREAD

**Discussion by Topping1** [ggml-org/llama.cpp#579] [Ideas]
The announcement is here:

https://www.cerebras.net/press-release/cerebras-systems-releases-seven-new-gpt-models-trained-on-cs-2-wafer-scale-systems

The models are available here:
https://huggingface.co/cerebras

Excerpts:
"SUNNYVALE, CALIFORNIA – March 28, 2023 – [Cerebras Systems](http://www.cerebras.net/), the pioneer in artificial intelligence (AI) compute for generative AI, today announced it has trained and is releasing a series of seven GPT-based large language models (LLMs) for open use by the research community. This is the first time a company has used non-GPU based AI systems to train LLMs up to 13 billion parameters and is sharing the models, weights, and training recipe via the industry standard Apache 2.0 license. All seven models were trained on the 16 CS-2 systems in the Cerebras Andromeda AI supercomputer."
"Cerebras’ release today directly addresses these issues. In a first among AI hardware companies, Cerebras researchers trained, on the Andromeda AI supercomputer, a series of seven GPT models with 111M, 256M, 590M, 1.3B, 2.7B, 6.7B, and 13B parameters."

**Comment by slaren** [comment]
Looking at the results from https://www.cerebras.net/blog/cerebras-gpt-a-family-of-open-compute-efficient-large-language-models/ I am not sure that these models are any better than GPT-J

**Comment by ggerganov** [comment]
Can someone link the python inference code if it is available - I cannot seem to find it

**Reply by slaren** [reply]
Looks like it is used the same way as any other HF model, there is an example at https://huggingface.co/cerebras/Cerebras-GPT-13B#quickstart

**Reply by ggerganov** [reply]
Yup, I just realized it. I have almost converted it

**Reply by ggerganov** [reply]
Ok, got it working:

<img width="1143" alt="image" src="https://user-images.githubusercontent.com/1991296/228359440-2d55ab81-f03f-49d4-bd68-44d5253876a7.png">

Will try to add a `ggml` example tomorrow

**Reply by appvoid** [reply]
Can you please give some insights on how you convert models? Is there a way to automate it or you need to know some structure behind the scenes? I'm new to all of this.

**Reply by ggerganov** [reply]
Here is the example with instructions:

https://github.com/ggerganov/ggml/tree/master/examples/gpt-2

**Reply by dogjamboree** [reply]
It works great, thanks! But for the 13B version the model is split into  2 huge bin files so the script doesn't know what to do with them. Any idea how to merge them for use?

**Comment by forrackun** [comment]
New 3B model looks quite capable https://www.cerebras.net/blog/btlm-3b-8k-7b-performance-in-a-3-billion-parameter-model/

**Comment by bornjre** [comment]
https://github.com/ggerganov/ggml/issues/427
